ui-v2/app/routes/dc/kv/index.js

```diff
@@ -13,7 +14,12 @@ export default Route.extend(WithKvActions, {
   },
   repo: service('kv'),
   model: function(params) {
-    const key = params.key || '/';
+    let key = params.key || '/';
+    // we are index or folder, so if the key doesn't have a trailing slash
+    // add one to force a fake findBySlug
+    if (!isFolder(key)) {
+      key = key + '/';
+    }
```

Comment:
Thinking about this some more - this solves the data fetching problem, but now you have 2 URLs that show the same thing which seems odd to me. What would you think about doing a `this.replaceWith` in the `beforeModel` hook that passes in a key with the slash on the end so that you always have a 1:1 mapping of URL to UI state?

Reply:
Took me a while to follow this, but this morning just had another look at that PR you sent over and I had a lightbulb moment! _ah ha, thats what matt was talking about in that comment!_ :smile: 

I thought I couldn't add the trailing slash because it would redirect again, but of course we are in ember by this point so we can redirect to a trailing slash using the history API and it won't be 301'ed

Good call, will sort in a bit hopefully be here by the time you are in 👍